SourceX/miniwin/misc_io.cpp

```diff
@@ -33,11 +33,13 @@ HANDLE CreateFileA(LPCSTR lpFileName, DWORD dwDesiredAccess, DWORD dwShareMode,
 	file->path = name;
 	if (dwCreationDisposition == DVL_OPEN_EXISTING) {
 		// read contents of existing file into buffer
-		std::ifstream filestream(file->path, std::ios::binary);
-		if (!filestream.fail()) {
-			file->buf.insert(file->buf.begin(),
-			    std::istreambuf_iterator<char>(filestream),
-			    std::istreambuf_iterator<char>());
+		std::ifstream filestream(file->path, std::ios::in | std::ios::binary);

```

Comment:
Open with `std::ios::ate` to remove the need for the first `seekg` call

Reply:
Both `seekg` calls?

Reply:
Only the first `seekg`. `std::ios::ate` takes you to the end of the file on opening.